api/handlers/roles.go

```diff
@@ -4,40 +4,24 @@ import (
 	"fmt"
 	"net"
 	"net/http"
-	"reflect"
 	"strings"
 
+	"github.com/globocom/gsh/api/auth"
 	"github.com/globocom/gsh/types"
 
 	"github.com/gosimple/slug"
 	"github.com/labstack/echo"
 )
 
-// getField returns the value of a field in a token or error if the field doesn't exist
-func getField(token *IDToken, field string) (string, error) {
-	r := reflect.ValueOf(token)
-	f := reflect.Indirect(r).FieldByName(field)
-	result := f.String()
-	if result == "<invalid Value>" {
-		return "", fmt.Errorf("getField: Field (%s) not found at IDToken", field)
-	}
-	return result, nil
-}
-
 // GetRolesForMe prints all the existing roles to current user
 func (h AppHandler) GetRolesForMe(c echo.Context) error {
 	// Validates JWT token before any other action
-	token, err := ValidateJWT(c, h.config)
+	ca := auth.OpenIDCAuth{}
+	username, err := ca.Authenticate(c, h.config)
 	if err != nil {
 		return c.JSON(http.StatusUnauthorized,
 			map[string]string{"result": "fail", "message": "Failed validating JWT", "details": err.Error()})
 	}
-	field := h.config.GetString("oidc_claim")
```

Comment:
Why did you remove this part of the code? The deletion is made several times in this file but I could get it yet. 🙃

Reply:
At the previous code, the function `ValidateJWT` returns the IDToken (and error). Using the IDToken, the next code gets the field with the username.
```
	username, err := getField(&token, field)	
	if err != nil {	
		return c.JSON(http.StatusInternalServerError,	
			map[string]string{"result": "fail", "message": "The field declared in oidc_claim doesn't exist", "details": err.Error()})	
	}
```
The new code `ca.Authenticate` returns the username (and error) and set as a context variable the JTI. This approach doesn't need the previous logic.

Reply:
Nice! I got your point! Good refactoring!